src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -236,17 +238,32 @@ def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
 
         # Use the query method to find rows that match the lookup parameters
         run_ids = pickle_metadata.query(query_conditions)["job_id"].to_list()
-        if run_ids:
-            filename = Path(backup_dir / run_ids[0]).with_suffix(".pkl")
-            with open(filename, "rb") as f:
+        possible_pickles = [
+            Path(backup_dir / run_id).with_suffix(".pkl") for run_id in run_ids
+        ]
+        existing_pickles = [pickle for pickle in possible_pickles if pickle.exists()]
+        if existing_pickles:
+            last_pickle = max(existing_pickles, key=os.path.getctime)
+            if len(existing_pickles) > 1:
+                logger.warning(
+                    f"Multiple backups found for {job_parameters}. Using the most recent and deleting the rest."
+                )
+                for stale_file in set(existing_pickles) - {last_pickle}:
+                    os.remove(stale_file)
+            with open(last_pickle, "rb") as f:
                 sim = dill.load(f)
+            current_job_id = get_current_job().id
+            logger.info(f"Renaming backup file {last_pickle} to {current_job_id}.pkl")
+            if job_parameters.backup_configuration["backup_freq"] is not None:
+                # Sleep to prevent FS latency when loading the pickle
+                sleep(5)
+                os.rename(last_pickle, (backup_dir / str(current_job_id)).with_suffix(".pkl"))
             return sim
-    except (OSError, FileNotFoundError):
+    except (OSError, FileNotFoundError, EOFError):
```

Comment:
When does the EOFError happen?

Reply:
In testing this, it seems that filesystem latency can cause a situation where we try to open a file that was modified (i think the os.rename()) recently enough that it is interpreted as an empty file. and unpiclking gives an EOFError. This is why I put the 5 second sleep in there, which seems to add enough time to avoid this.

I also wanted to catch situations where the pickle file is corrupted, e.g. if the sim crashes "just" during the write of the pickle, I guess it might make sense to try a few attempts before moving on to just restarting the sim.